realm/realm-annotations-processor/src/test/resources/io/realm/AllTypesRealmProxy.java

```diff
@@ -170,6 +171,9 @@ private void injectObjectContext() {
             if (!proxyState.getAcceptDefaultValue$realm()) {
                 return;
             }
+            final Row row = proxyState.getRow$realm();
+            row.getTable().setLong(columnInfo.columnLongIndex, row.getIndex(), value, true);
```

Comment:
That would actually be the better design longterm, short term I was mostly thinking that we expose it in the row interface, since Unchecked row already have an reference to the table, we could expose `setDefaultString(index, val)` and friends, but it would inflate the method count, so perhaps it is just better to wait for core support.

For now, I think it is fine to just create a Core ticket requesting this.